src/code.cloudfoundry.org/gorouter/route/pool.go

```diff
@@ -74,6 +74,21 @@ type ProxyRoundTripper interface {
 	CancelRequest(*http.Request)
 }
 
+type HashRoutingProperties struct {
```

Comment:
I'm not sure if I agree with splitting this into a separate struct. If we were to do this it would probably be more consistent to have a *Options struct to keep some resemblance to the structure of the message. Alternative is to put the fields directly into the Endpoint.

Edit: especially since the EndpointOpts does exactly that and it would remove the risk of a bunch of nil pointer panics.

Reply:
Checking which algorithm is used is done via `LoadBalancingAlgorithm`, no? And if it is set to `hash` the rest of the code can assume that the required properties were checked during the endpoint creation and can safely be used.